File: TopInterview150/167_two_sum_sorted/two_sum.go
package twosumsorted

func twoSum(numbers []int, target int) []int {
	l, r := 0, len(numbers)-1

	for l < r {
		if numbers[l]+numbers[r] == target {
			return []int{l + 1, r + 1}
		}
		if numbers[l]+numbers[r] > target {
			r = r - 1
		}
		if numbers[l]+numbers[r] < target {
			l = l + 1
		}
	}

	return []int{l, r}
}

File: TopInterview150/167_two_sum_sorted/TwoSum.cs
public class TwoSum {
    public int[] TwoSum(int[] numbers, int target) {
        int l = 0;
        int r = numbers.Length - 1;

        while (l < r) {
            if (numbers[l] + numbers[r] == target) {
                return new int[] { l + 1, r + 1 };
            }

            if (numbers[l] + numbers[r] > target) {
                r--;
            }

            if (numbers[l] + numbers[r] < target) {
                l++;
            }
        }

        return new int[] {};
    }
}